lib/base/utility.hpp

```diff
@@ -197,6 +202,47 @@ class Utility
 	static boost::thread_specific_ptr<unsigned int> m_RandSeed;
 };
 
+template<class Fstream>
+String Utility::CreateTempFile(const String& path, int mode, Fstream& fp)
+{
+	std::vector<char> targetPath(path.Begin(), path.End());
+	targetPath.push_back('\0');
```

Comment:
By the way, this is no longer needed with new guarantees for `std::string::data()` in C++11.